middleware/src/middleware_test.go

```diff
@@ -236,106 +237,103 @@ func TestSetRootPassword(t *testing.T) {
+	authenticateAdminChanged := testMiddleware.UserAuthenticate(adminChangedArgs)
 
-	require.Equal(t, true, authenticateDev.Success)
-	require.Equal(t, true, testMiddleware.DummyIsBaseSetup())
+	require.Equal(t, true, authenticateAdminChanged.ErrorResponse.Success)
+	require.NoError(t, testMiddleware.ValidateToken(authenticateAdmin.Token))
+	setupStatus = testMiddleware.SetupStatus()
+	require.Equal(t, true, setupStatus.MiddlewarePasswordSet)
 
-	/* test initial admin login, should fail now because dummyIsBaseSetup == true  */
+	/* test initial admin login, should fail now because isMiddlewarePasswordSet == true  */
 	authenticateAdmin2 := testMiddleware.UserAuthenticate(adminArgs)
 
-	require.Equal(t, false, authenticateAdmin2.Success, false)
-	require.Equal(t, "authentication unsuccessful", authenticateAdmin2.Message)
-	require.Equal(t, true, testMiddleware.DummyIsBaseSetup(), true)
-
-	/* test initial admin login with dummyAdminPassword. Should succeed now, because dummyIsBaseSetup == true  */
-	adminDummyPW2Args := rpcmessages.UserAuthenticateArgs{Username: "admin", Password: testMiddleware.DummyAdminPassword()}
-	authenticateAdminDummyPW2 := testMiddleware.UserAuthenticate(adminDummyPW2Args)
-
-	require.Equal(t, true, authenticateAdminDummyPW2.Success)
-	require.Equal(t, true, testMiddleware.DummyIsBaseSetup())
+	require.Equal(t, false, authenticateAdmin2.ErrorResponse.Success, false)
+	require.Equal(t, "authentication unsuccessful, incorrect password", authenticateAdmin2.ErrorResponse.Message)
+	require.Equal(t, "", authenticateAdmin2.Token)
+	setupStatus = testMiddleware.SetupStatus()
+	require.Equal(t, true, setupStatus.MiddlewarePasswordSet, true)
 
 	/* test invalid login (with a invalid username) */
 	invalidNameArgs := rpcmessages.UserAuthenticateArgs{Username: "InvalidUserName", Password: ""}
 	authenticateInvalidName := testMiddleware.UserAuthenticate(invalidNameArgs)
 
-	require.Equal(t, false, authenticateInvalidName.Success)
-	require.Equal(t, "authentication unsuccessful", authenticateInvalidName.Message)
+	require.Equal(t, false, authenticateInvalidName.ErrorResponse.Success)
+	require.Equal(t, "authentication unsuccessful, username not found", authenticateInvalidName.ErrorResponse.Message)
 
 	/* test invalid login (empty username and password) */
 	emptyArgs := rpcmessages.UserAuthenticateArgs{Username: "", Password: ""}
 	authenticateEmpty := testMiddleware.UserAuthenticate(emptyArgs)
 
-	require.Equal(t, false, authenticateEmpty.Success)
-	require.Equal(t, "authentication unsuccessful", authenticateEmpty.Message)
+	require.Equal(t, false, authenticateEmpty.ErrorResponse.Success)
+	require.Equal(t, "authentication unsuccessful, username not found", authenticateEmpty.ErrorResponse.Message)
 }
 
 func TestUserChangePassword(t *testing.T) {
-	testMiddleware := setupTestMiddleware()
+	testMiddleware := setupTestMiddleware(t)
 
 	/* test valid password change */
```

Comment:
comment needs to be adjusted to code or removed

Reply:
adjusted.